src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java

```diff
@@ -60,19 +61,24 @@ public NtlmAuthenticator create() {
         }
     }
 
-    public long authenticate(Connection connection, AuthenticationContext context) throws TransportException {
+    public Session authenticate(Connection connection, AuthenticationContext context) throws TransportException {
```

Comment:
Still I think the Authenticator is the wrong place to create the session. The connection builds the session, the authentication takes care of authenticating the session. One other option is that we pass in an `UnauthenticatedSession` object that constructs the 'real' `Session`. Instead of the `AuthenticationContext` that we now pass in.

The fact that you had to 'open up' classes that were made package private is to me always an immediate smell.